src/OpenTelemetry.OpAMPClient/FrameProcessor.cs

```diff
@@ -3,17 +3,40 @@
 
 using System.Buffers;
 using Opamp.Protocol;
+using OpenTelemetry.Internal;
+using OpenTelemetry.OpAMPClient.Listeners;
+using OpenTelemetry.OpAMPClient.Listeners.Messages;
 using OpenTelemetry.OpAMPClient.Utils;
 
 namespace OpenTelemetry.OpAMPClient;
 
 internal class FrameProcessor
 {
-    private readonly IOpAMPMessageListener listener;
+    private readonly Dictionary<Type, List<IOpAMPListener>> listeners = [];
```

Comment:
Do we need to consider using concurrent types here for thread safety?

Reply:
true, might not be so critical for SDK but Auto inst could initialize and subscribe late. fill fix it now.

Reply:
actually now thinking that even opamp messages could trigger new services that can listen and unsubscribe... so still critical.